contrib/CODINGSTYLE.md

```diff
@@ -0,0 +1,43 @@
+# Coding Style
+## Introduction
+This is the coding style guide for Modmesh.  
+* [Introduction](#introduction)
+* [Naming](#naming)
+* [Comments](#comments)
+* [Formatting](#formatting)
+* [Class](#class)
+* [Functions](#functions)
+
+### Naming
+1. The class names use Camel case, such as ```class R3DWidget;```. 
+2. The name of the variable / using-declarations uses snake case,  
+such as ```R3DWidget * viewer;``` and ```using size_type = std::size_t;```. 
+3. The QT-related function name uses Camel case,  
```

Comment:
what's the difference with point 1.

Reply:
I notice that whether the class is related QT, the naming of the class is Camel case.

Reply:
Class names are mostly CamelCase, although exceptions are not rare when it comes to a helper class.

Reply:
> helper class

What is the style of the helper class? 
Should I mention the helper class in the markdown?

Reply:
It's complicated

Reply:
> It's complicated

Ok, we may skip the helper class temporaily.